src/libsync/determineuserjobfactory.cpp

```diff
@@ -0,0 +1,61 @@
+#include "determineuserjobfactory.h"
+#include "common/utility.h"
+#include "creds/httpcredentials.h"
+#include <QJsonParseError>
+#include <QNetworkReply>
+
+Q_LOGGING_CATEGORY(lcDetermineUserJob, "sync.networkjob.determineuserjob", QtInfoMsg);
+
+using namespace OCC;
+
+DetermineUserJobFactory::DetermineUserJobFactory(QNetworkAccessManager *networkAccessManager, const QString &accessToken, QObject *parent)
```

Comment:
That one is a bit different, its a oc json job, it always has the account url as base and we should usually use the JsonApiJob class for it to ensure proper error handling.
As we need a non account bound version only in the Oath flow, I don't think we should make this class generally available.

Reply:
I'll move it there. Long term goal should be to eliminate the duplicate code, though. I could imagine that we can have a shared code base for both jobs, one of which is extended with account-related functionality. We just did something similar in https://github.com/owncloud/client/pull/9422.